src/python/pants/backend/python/util_rules/vcs_versioning.py

```diff
@@ -99,8 +111,49 @@ async def generate_python_from_setuptools_scm(
     return GeneratedSources(output_snapshot)
 
 
+# This is only used to register our implementation with the plugin hook via unions.
+class PythonVCSVersionMappingMarker(FirstPartyPythonMappingImplMarker):
+    pass
+
+
+class VCSVersionPythonResolveField(PythonResolveField):
+    alias = "python_resolve"
+
+
+@rule
+async def map_to_python_modules(
+    all_targets: AllTargets,
+    python_setup: PythonSetup,
+    _: PythonVCSVersionMappingMarker,
+) -> FirstPartyPythonMappingImpl:
+    suffix = ".py"
```

Comment:
Maybe worth mentioning this expectation of `.py` in the `help` message?

Reply:
That's the thing - there isn't really such an expectation! For example, you can generate a .txt file and have multiple languages read from it. Obviously dep inference won't work in that case, nor would anyone expect it to...